packages/ui/src/components/Designer/RightSidebar/DetailView/index.tsx

```diff
@@ -66,14 +66,29 @@ const DetailView = (props: DetailViewProps) => {
     values.y = values.position.y;
     delete values.position;
 
-    if (values.key !== (form.getValues() || {}).key) {
```

Comment:
move this to the `useEffect` below due to an edge case. If you change the key to that of another schema then select the new schema of the same key it wouldn't reset.